File: Sample Algorithms/Exam Study/Recursion Exercies/1.c
// Write a recursive program that calculates the sum of all the elements in an array

#include <stdio.h>


int calcSum(int array[],int n){
    if (n < 0){
        return 0;
    }
    return array[n] + calcSum(array,n-1);
}

void main() {
    int a[] = {1,2,3};
    int sum = calcSum(a,sizeof(a)/sizeof(a[0])-1);
    printf("%d",sum);
}

File: Sample Algorithms/Exam Study/Recursion Exercies/1.py
# Write a recursive program that calculates the sum of all the elements in an array

def calcSum(arr,n):
    if n < 0:
        return 0
    return arr[n] + calcSum(arr,n-1)


a = [1,2,3,4,5,6]
print (calcSum(a,len(a)-1))